packages/node/src/indexer/apiPromise.connection.ts

```diff
@@ -53,6 +60,50 @@ export class ApiPromiseConnection implements ApiConnection {
     await this._api.disconnect();
   }
 
+  /* eslint-disable prefer-rest-params */
+  static createCachedProvider(provider: WsProvider | HttpProvider) {
```

Comment:
I don't think this needs to be static. It can be a function outside of a class.

```suggestion
  static createCachedProvider(provider: ProviderInterface): ProviderInterface {
```